### ISSUE TITLE
RC: On some installs chromium-kiosk&&qiosk boots up sooner than internet connection is established

### ISSUE THREAD

**Issue by Salamek** [Salamek/chromium-kiosk#88]
This causes the browser to show "No internet..." 

Solution add connection check to requested URL before starting qiosk in chromium-kiosk script

OR in qiosk on QWebEngine network error run something like:

```c++
MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), currentInterval(initialInterval) {

    webEngineView = new QWebEngineView(this);
    webEngineView->setUrl(QUrl("http://example.com"));

    networkManager = new QNetworkAccessManager(this);
    connect(networkManager, &QNetworkAccessManager::finished, this, &MainWindow::handleNetworkReply);

    timer = new QTimer(this);
    connect(timer, &QTimer::timeout, this, &MainWindow::checkUrlAccessibility);
    timer->start(currentInterval);  // Start with initial interval

    connect(webEngineView, &QWebEngineView::loadFinished, this, &MainWindow::handleLoadFinished);

    QWidget *centralWidget = new QWidget(this);
    QVBoxLayout *layout = new QVBoxLayout(centralWidget);
    layout->addWidget(webEngineView);
    setCentralWidget(centralWidget);

    connect(this, &MainWindow::urlAccessible, this, [this]() {
        qDebug() << "URL is accessible. Reloading the page.";
        webEngineView->reload();
        currentInterval = initialInterval; // Reset interval on success
        timer->start(currentInterval);
    });
}

MainWindow::~MainWindow() {
    // Destructor
}

void MainWindow::checkUrlAccessibility() {
    QNetworkRequest request(QUrl("http://example.com"));
    networkManager->get(request);
}

void MainWindow::handleNetworkReply(QNetworkReply *reply) {
    if (reply->error() == QNetworkReply::NoError) {
        emit urlAccessible();
    } else {
        qDebug() << "URL is not accessible.";
        if (currentInterval < maxInterval) {
            currentInterval = qMin(currentInterval + incrementStep, maxInterval);
        }
        timer->start(currentInterval); // Increment the interval
    }
    reply->deleteLater();
}

void MainWindow::handleLoadFinished(bool success) {
    if (!success) {
        qDebug() << "Load finished with error. Checking URL accessibility.";
        checkUrlAccessibility();
    }
}
```

in a loop

**Comment by mishoboss** [comment]
Hey Adam, glad you found the issue. I'm doing a nasty workaround to make this work since several months (a script that changes the URL in config 10 seconds after boot).

Do you have a roadmap for this fix?

**Comment by Salamek** [comment]
@mishoboss fix is actually in qiosk master right now, it just needs to be properly tested before release...

**Comment by mishoboss** [comment]
I could test it on my setup if you give me a x86 deb file.

**Comment by Salamek** [comment]
@mishoboss already released in `chromium-kiosk` 2.5 and `qiosk`  1.8, check it out
